### PR TITLE
fix: cowswap receive amount

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Fixes a bug where we double counting the CoW Swap fee, which was causing us to show lower-than-actual receive amounts for CoW Swap quotes, and impacting minimum amount UI.

Also improves naming throughout to better convey the data each constant holds.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Small/medium. If CoW Swap quotes look sane, we should be gravy.

## Testing

Check our CoW Swap quotes against https://swap.cow.fi/ and confirm that they are similar (they will be slightly different due to rounding/fiat conversions etc).

Note, the `buyAmount` in the response from the `https://api.cow.fi/mainnet/api/v1/quote` endpoint sometimes varies wildly with the same request args, which is why the amount sometimes changes significantly between the quote screen and the trade confirm screen.

### Engineering

☝️

Check the response to the `https://api.cow.fi/mainnet/api/v1/quote` POST request (the "sell" kind).

The `buyAmount` field should roughly match that which we are showing in the "You get" amount. 

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-05-08 at 8 29 58 pm

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4448** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4448" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4448?utm_source=stack-comment).

File: src/lib/swapper/swappers/CowSwapper/cowBuildTrade/cowBuildTrade.test.ts [hunk 1]
```diff
@@ -132,59 +132,59 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
     sellAssetTradeFeeUsd: '17.95954294012756741283729339486489192096',
   },
   sellAmountBeforeFeesCryptoBaseUnit: '1000000000000000000',
-  buyAmountCryptoBaseUnit: '14501811818247595090576', // 14501 FOX
+  buyAmountBeforeFeesCryptoBaseUnit: '14707533959600717283163', // 14501 FOX
```
Comment (woodenfurniture) [inline src/lib/swapper/swappers/CowSwapper/cowBuildTrade/cowBuildTrade.test.ts:135]:
```suggestion
  buyAmountBeforeFeesCryptoBaseUnit: '14707533959600717283163', // 14707 FOX
```

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
seeing a couple issues with the amounts when running locally
- gas fee appearing as 0
- it seems that the gas fee is being subtracted from the sell amount on cowswap's side -- `47914924+34726523 = 82641447`, but the other swappers add the gas fee separately

image

**Comment by 0xApotheosis** [comment]
The gas fee on CoW swaps _is_ 0, ser 👌

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**